eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -228,43 +244,52 @@ class Peer(nodeParams: NodeParams, remoteNodeId: PublicKey, authenticator: Actor
       log.info(s"sent all announcements to {}: channels={} updates={} nodes={}", remoteNodeId, channelsSent, updatesSent, nodesSent)
       stay
 
-    case Event(Rebroadcast(channels, updates, nodes), ConnectedData(_, transport, _, _)) =>
-      // we filter out announcements that we received from this node
-      def sendIfNeeded(m: Map[_ <: LightningMessage, Set[ActorRef]]): Int = m.foldLeft(0) {
-        case (counter, (_, origins)) if origins.contains(self) =>
-          counter // won't send back announcement we received from this peer
-        case (counter, (announcement, _)) =>
-          transport ! announcement
-          counter + 1
-      }
-      val channelsSent = sendIfNeeded(channels)
-      val updatesSent = sendIfNeeded(updates)
-      val nodesSent = sendIfNeeded(nodes)
+    case Event(rebroadcast: Rebroadcast, ConnectedData(_, transport, _, _, maybeGossipTimestampFilter)) =>
+      val (channels1, updates1, nodes1) = Peer.filterGossipMessages(rebroadcast, self, maybeGossipTimestampFilter)
+      val channelsSent = channels1.length
```

Comment:
I'd rather do that in a `fold` (like it was before) so that we don't iterate twice